js/capture.js

```diff
@@ -19,13 +27,61 @@ const capture = {
       });
   },
 
+  // In the parsed whitelist JSON object (from disconnect-entitylist.json):
+  //   - 'properties' keys are first parties
+  //   - 'resources' keys are third parties
+  reformatList(whiteList) {
+    const firstPartyWhiteList = {};
+    const thirdPartyWhiteList = {};
+    let counter = 0;
+    for (const siteOwner in whiteList) {
+      const firstParties = whiteList[siteOwner].properties;
+      for (let i = 0; i < firstParties.length; i++) {
+        firstPartyWhiteList[firstParties[i]] = counter;
+      }
+      const thirdParties = whiteList[siteOwner].resources;
+      thirdPartyWhiteList[counter] = [];
+      for (let i = 0; i < thirdParties.length; i++) {
+        thirdPartyWhiteList[counter].push(thirdParties[i]);
+      }
+      counter++;
+    }
+
+    return {
+      firstPartyWhiteList,
+      thirdPartyWhiteList
+    };
+  },
+
   // Returns true if the request should be stored, otherwise false.
-  shouldStore(tab) {
+  shouldStore(tab, thirdPartyFromRequest) {
     const documentUrl = new URL(tab.url);
     // ignore about:*, moz-extension:* & non-visible tabs (like dev tools)
     if (documentUrl.protocol !== 'about:'
       && documentUrl.protocol !== 'moz-extension:'
       && tab.id !== browser.tabs.TAB_ID_NONE) {
+      // check if third party is on the whitelist (owned by the first party)
+      if (thirdPartyFromRequest) {
+        const firstPartyFromRequest = documentUrl.hostname;
+        for (const firstParty in this.firstPartyWhiteList) {
+          const firstPartyFromList = new RegExp(firstParty);
```

Comment:
> I would rather hope we could just duplicate google.com to www.google.com in the list and use an object key access and array includes rather than looping through all first parties if we can help it.

The hostnames from the third party requests are pretty gnarly. Examples include `r5---sn-4g5e6n76.googlevideo.com`, and the first parties' hostnames could be something like `developer.mozilla.com` with or without a subdomain. On the other end, first and third parties in the whitelist can be of the form `example.com` or `www.example.com`. I don't know all the possible variations on inputs from the requests and how to standardize these to use a direct string comparison using key and array access like `object.hasOwnProperty` or `array.includes`, particularly when the Disconnect list itself is inconsistent in its use of subdomains (`www` and even found a `www.hands.com.br`). That's why I used regExp.  If you have ideas, please share!